src/core/public/theme/core_theme_provider.tsx

```diff
@@ -21,16 +23,26 @@ interface CoreThemeProviderProps {
   theme$: Observable<CoreTheme>;
 }
 
+const emotionCache = createCache({
+  key: 'eui-styles',
+  container: document.querySelector(`meta[name="${EUI_STYLES_GLOBAL}"]`) as HTMLElement,
+});
+
 /**
- * Wrapper around `EuiThemeProvider` converting (and exposing) core's theme to EUI theme.
+ * Wrapper around `EuiProvider` converting (and exposing) core's theme to EUI theme.
  * @internal Only meant to be used within core for internal usages of EUI/React
  */
 export const CoreThemeProvider: FC<CoreThemeProviderProps> = ({ theme$, children }) => {
   const coreTheme = useObservable(theme$, defaultTheme);
   const euiTheme = useMemo(() => convertCoreTheme(coreTheme), [coreTheme]);
   return (
-    <EuiThemeProvider colorMode={euiTheme.colorMode} theme={euiTheme.euiThemeSystem}>
+    <EuiProvider
+      modify={{ LIGHT: { colors: { primary: '#FF0000' } } }}
```

Comment:
I think declaring a const outside of the component is slightly better for memory and rerendering triggers as it would  not be recreated at each rerendering + only one instance will be used for all usages of `CoreThemeProvider`

Reply:
Thanks for flagging this. This was meant entirely to test the setup and will be removed before merging.

Reply:
We will want to expand the API to pass `modify` to KibanaThemeProvider, but I don't see that as an immediate need.